code/modules/recycling/sortingmachinery.dm

```diff
@@ -230,45 +230,60 @@
 	desc = "Used to set the destination of properly wrapped packages."
 	icon = 'icons/obj/device.dmi'
 	icon_state = "dest_tagger"
-	var/currTag = 0
-	//The whole system for the sorttype var is determined based on the order of this list,
-	//disposals must always be 1, since anything that's untagged will automatically go to disposals, or sorttype = 1 --Superxpdude
 
 	w_class = WEIGHT_CLASS_TINY
 	item_state = "electronic"
 	flags = CONDUCT
 	slot_flags = SLOT_BELT
-
-/obj/item/destTagger/proc/openwindow(mob/user as mob)
-	var/dat = "<tt><center><h1><b>TagMaster 2.2</b></h1></center>"
-
-	dat += "<table style='width:100%; padding:4px;'><tr>"
-	for(var/i = 1, i <= GLOB.TAGGERLOCATIONS.len, i++)
-		dat += "<td><a href='?src=[UID()];nextTag=[i]'>[GLOB.TAGGERLOCATIONS[i]]</a></td>"
-
-		if(i%4==0)
-			dat += "</tr><tr>"
-
-	dat += "</tr></table><br>Current Selection: [currTag ? GLOB.TAGGERLOCATIONS[currTag] : "None"]</tt>"
-
-	user << browse(dat, "window=destTagScreen;size=450x350")
-	onclose(user, "destTagScreen")
+	///Value of the tag
+	var/currTag = 0
+	//The whole system for the sorttype var is determined based on the order of this list,
+	//disposals must always be 1, since anything that's untagged will automatically go to disposals, or sorttype = 1 --Superxpdude
 
 /obj/item/destTagger/attack_self(mob/user as mob)
-	openwindow(user)
-	return
+	ui_interact(user)
+
+/obj/item/destTagger/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "DestinationTagger", name, 395, 350, master_ui, state)
+		ui.open()
+
+/obj/item/destTagger/ui_data(mob/user)
+	var/list/data = list()
+
+	data["destinations"] = list()
+	data["selected_destination"] = list()
+	var/destination_index = 1
+	for(var/destination in GLOB.TAGGERLOCATIONS)
+		var/list/destination_data = list(
+			"name" = destination,
+			"id"   = destination_index,
+		)
+		data["destinations"] += list(destination_data)
+		if(destination_index == currTag)
+			data["selected_destination"] = destination_data
+		destination_index++
```

Comment:
This sort of iteration `destination_index` increment is very prone to errors. Make it be part of the `for` , that is
```dm
for(var/destination_index in 1 to length(GLOB.TAGGERLOCATIONS))
  var/destination = GLOB.TAGGERLOCATIONS[destination_index]
  // rest of the code ..
```

Yes, we do prefer the `for(var/foo in my_list)` over `for(var/i = 1; i <= length(my_list); i++ ){var/foo=my_list[i]; ... }` usually, but that's because most of the time we don't care about the index, and only want list contents. This is not one of those cases though.